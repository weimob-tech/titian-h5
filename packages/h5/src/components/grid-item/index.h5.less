@import '../common/static/var';
@import '../common/static/utils';
@import '../common/static/mixins/index.less';

@gridItem: ~'@{project}-grid-item';

.@{gridItem} {
  display: block;
  position: relative;
  box-sizing: border-box;

  &-host {
    display: contents;
  }

  &-square {
    height: 0;
  }

  &-content {
    display: flex;
    box-sizing: border-box;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100%;
    padding: @grid-item-padding-v @grid-item-padding-h;

    background-color: @grid-item-bg-color;

    &-square {
      position: absolute;
      top: 0;
      right: 0;
      left: 0;
    }

    &::before {
      .hairline(right);
    }

    &::after {
      .hairline(bottom);
    }

    &-gutter,
    &-borderless {
      &::before,
      &::after {
        content: none;
      }
    }
  }

  &-text {
    padding: @grid-text-padding-v @grid-text-padding-h;

    color: @grid-item-text-color;
    font-size: @grid-item-text-font-size;
    line-height: @grid-item-text-line-height;
  }
}
