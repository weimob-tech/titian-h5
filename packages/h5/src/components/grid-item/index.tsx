import { Component, h, Host, Method, Prop, State, Element } from '@stencil/core';
import { BasicComponentAbstract } from '../common/basic/BasicComponent';
import { isString, addShadowRootStyle } from '../common/utils';
import * as namespace from '../common/utils/namespace';
import { getParent } from '../common/utils/relation';
import addUnit from '../common/utils/suffix';
import { TiGrid } from '../grid';

@Component({
  tag: 'ti-grid-item',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
  shadow: true,
})
export class TiGridItem implements BasicComponentAbstract {
  @Element() host: HTMLElement;

  @Prop() extCss = '';

  @Prop() extClass?: string;

  @Prop() contentClass?: string;

  @Prop() iconClass?: string;

  @Prop() textClass?: string;

  /**
   * 图标
   *
   * @type string
   * @default ''
   * @example
   * <TiGridItem icon="home" text="Grid" />
   * @since 0.1.0
   */
  @Prop() icon?: string;

  /**
   * 文字内容
   *
   * @type string
   * @default ''
   * @example
   * <TiGridItem icon="home" text="Grid" />
   * @since 0.1.0
   */
  @Prop() text?: string;

  /**
   * 文字颜色
   *
   * @type string
   * @default ''
   * @example
   * <TiGridItem icon="home" text="Grid" color="red" />
   * @since 0.1.0
   */
  @Prop() color?: string;

  /**
   * 图标尺寸
   *
   * @type number
   * @default ''
   * @example
   * <TiGridItem icon="home" text="Grid" size={{32}} />
   * @since 0.1.0
   */
  @Prop() size?: number;

  /**
   * 是否自定义内容
   *
   * @type boolean
   * @default false
   * @example
   * <TiGridItem customContent />
   * @since 0.1.0
   */
  @Prop() customContent?: boolean = false;

  @State() direction: 'row' | 'column' = 'column';

  @State() square = false;

  @State() divider = true;

  @State() columns = 4;

  @State() gutter = 0;

  get parent() {
    return getParent({
      host: this.host,
      tag: 'ti-grid',
      relations: 'parent',
    })?.[0];
  }

  async componentWillLoad(): Promise<void> {
    addShadowRootStyle.call(this);

    await this.updateDataFromParent();
  }

  @Method()
  async updateDataFromParent(parent?: HTMLTiGridElement | TiGrid) {
    if (!parent) {
      parent = this.parent;
    }
    this.gutter = parent.gutter;
    this.square = parent.square;
    this.divider = parent.divider;
    this.columns = parent.columns;
    this.direction = parent.direction;
  }

  private getClassName() {
    const { extClass, square } = this;
    const classList = [namespace.join('grid-item', { square })];
    if (extClass) {
      classList.push(extClass);
    }
    return classList.join(' ');
  }

  private renderIcon() {
    const { icon, size, color } = this;

    if (isString(icon) && icon) {
      return (
        <div part={this.iconClass} class={`${namespace.join('grid-item-icon')} ${this.iconClass}`}>
          {<ti-icon name={icon as string} size={size} color={color} /> || null}
        </div>
      );
    }

    return <slot name="icon" />;
  }

  private renderText() {
    const { text } = this;

    if (isString(text) && text) {
      return (
        <div part={this.textClass} class={`${namespace.handle('grid-item', 'text')} ${this.textClass}`}>
          {text || null}
        </div>
      );
    }

    return <slot name="text" />;
  }

  private renderContent() {
    const { customContent } = this;
    if (customContent) {
      return <slot name="content" />;
    }

    return [this.renderIcon(), this.renderText()];
  }

  private completedStyles() {
    const { columns, square, gutter } = this;
    const width = `${100 / columns}%`;

    const styles = {
      width,
    };

    if (square) {
      styles['padding-top'] = width;
    }

    if (gutter) {
      const gutterWidth = addUnit(gutter / 2);
      styles['margin-top'] = gutterWidth;
      styles['padding-right'] = gutterWidth;
    }

    return styles;
  }

  render() {
    const { square, divider, gutter, direction } = this;

    return (
      <Host style={{ display: 'contents' }}>
        <div part={this.extClass} class={this.getClassName()} style={this.completedStyles()}>
          <div
            part={this.contentClass}
            class={`${namespace.join('grid-item-content', { square, borderless: !divider, gutter: gutter > 0 })} ${
              this.contentClass
            }`}
            style={direction ? { flexDirection: direction } : {}}
          >
            {this.renderContent()}
          </div>
        </div>
      </Host>
    );
  }
}
