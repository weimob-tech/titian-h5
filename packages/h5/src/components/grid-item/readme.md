# ti-grid-item



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description      | Type      | Default     |
| --------------- | ---------------- | ---------------- | --------- | ----------- |
| `color`         | `color`          | 文字颜色             | `string`  | `undefined` |
| `contentClass`  | `content-class`  |                  | `string`  | `undefined` |
| `customContent` | `custom-content` | 是否自定义内容          | `boolean` | `false`     |
| `extClass`      | `ext-class`      | 额外的类名，添加到根节点的元素上 | `string`  | `undefined` |
| `icon`          | `icon`           | 图标               | `string`  | `undefined` |
| `iconClass`     | `icon-class`     |                  | `string`  | `undefined` |
| `size`          | `size`           | 图标尺寸             | `number`  | `undefined` |
| `text`          | `text`           | 文字内容             | `string`  | `undefined` |
| `textClass`     | `text-class`     |                  | `string`  | `undefined` |


## Methods

### `updateDataFromParent(parent?: HTMLTiGridElement | Components.TiGrid) => Promise<void>`



#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [ti-icon](../icon)

### Graph
```mermaid
graph TD;
  ti-grid-item --> ti-icon
  style ti-grid-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
