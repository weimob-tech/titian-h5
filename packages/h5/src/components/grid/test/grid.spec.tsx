import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TiGrid } from '../index';
import { TiGridItem } from '../../grid-item';

describe('渲染宫格组件', () => {
  it('基本使用', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid>
          <ti-grid-item extClass="custom-item" icon="mine-to-pay" text="待付款" />
          <ti-grid-item icon="to-deliver" text="待收货" />
          <ti-grid-item icon="to-receive" text="待发货" />
          <ti-grid-item icon="to-comment" text="待评价" />
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('自定义每行个数', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid columns={3}>
          <ti-grid-item icon="mine-to-pay" text="待付款" />
          <ti-grid-item icon="to-deliver" text="待收货" />
          <ti-grid-item icon="to-receive" text="待发货" />
          <ti-grid-item icon="to-comment" text="待评价" />
          <ti-grid-item icon="to-refund" text="退货" />
          <ti-grid-item icon="camera-point" text="拍照展示" />
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
    page.root.setAttribute('columns', '4');
  });

  it('设置宫格间距离', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid gutter={16}>
          <ti-grid-item icon="mine-to-pay" text="待付款" />
          <ti-grid-item icon="to-deliver" text="待收货" />
          <ti-grid-item icon="to-receive" text="待发货" />
          <ti-grid-item icon="to-comment" text="待评价" />
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('自适应展示正方型', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid square>
          <ti-grid-item icon="mine-to-pay" text="待付款" />
          <ti-grid-item icon="to-deliver" text="待收货" />
          <ti-grid-item icon="to-receive" text="待发货" />
          <ti-grid-item icon="to-comment" text="待评价" />
          <ti-grid-item icon="to-refund" text="退货" />
          <ti-grid-item icon="camera-point" text="拍照展示" />
          <ti-grid-item icon="to-deliver" text="待收货" />
          <ti-grid-item icon="to-receive" text="待发货" />
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('自定义图标尺寸，颜色', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid title="自定义图标尺寸，颜色">
          <ti-grid-item icon="mine-to-pay" size={60} text="图标尺寸" />
          <ti-grid-item icon="to-deliver" color="red" text="定义颜色" />
          <ti-grid-item icon="to-receive" text="待发货" />
          <ti-grid-item icon="to-comment" text="待评价" />
          <ti-grid-item icon="to-refund" text="退货" />
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('使用 slot 定义内容', async () => {
    const page = await newSpecPage({
      components: [TiGrid, TiGridItem],
      template: () => (
        <ti-grid title="使用 slot 定义内容">
          <ti-grid-item custom-content>
            <div slot="content">待付款</div>
          </ti-grid-item>
          <ti-grid-item>
            <div slot="icon"></div>
            <div slot="text">待收货</div>
          </ti-grid-item>
          <ti-grid-item custom-content>
            <div slot="content">待发货</div>
          </ti-grid-item>
          <ti-grid-item custom-content>
            <div slot="content">待评价</div>
          </ti-grid-item>
        </ti-grid>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
