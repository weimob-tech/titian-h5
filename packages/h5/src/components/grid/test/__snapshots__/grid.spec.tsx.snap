// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染宫格组件 使用 slot 定义内容 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <div class="titian-grid titian-grid-title">
        使用 slot 定义内容
      </div>
      <div class="titian-grid titian-grid-divider">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item custom-content="" style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <slot name="content"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="content">
      待付款
    </div>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <slot name="icon"></slot>
          <slot name="text"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="icon"></div>
    <div slot="text">
      待收货
    </div>
  </ti-grid-item>
  <ti-grid-item custom-content="" style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <slot name="content"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="content">
      待发货
    </div>
  </ti-grid-item>
  <ti-grid-item custom-content="" style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <slot name="content"></slot>
        </div>
      </div>
    </mock:shadow-root>
    <div slot="content">
      待评价
    </div>
  </ti-grid-item>
</ti-grid>
`;

exports[`渲染宫格组件 基本使用 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <slot name="title"></slot>
      <div class="titian-grid titian-grid-divider">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="custom-item titian-grid-item" part="custom-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="mine-to-pay"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待付款
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待收货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-comment"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待评价
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
</ti-grid>
`;

exports[`渲染宫格组件 自定义图标尺寸，颜色 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <div class="titian-grid titian-grid-title">
        自定义图标尺寸，颜色
      </div>
      <div class="titian-grid titian-grid-divider">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="mine-to-pay" size="60"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            图标尺寸
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon color="red" name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            定义颜色
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-comment"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待评价
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-refund"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            退货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
</ti-grid>
`;

exports[`渲染宫格组件 自定义每行个数 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <slot name="title"></slot>
      <div class="titian-grid titian-grid-divider">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="mine-to-pay"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待付款
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待收货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-comment"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待评价
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-refund"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            退货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 33.333333333333336%;">
        <div class="titian-grid-item-content undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="camera-point"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            拍照展示
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
</ti-grid>
`;

exports[`渲染宫格组件 自适应展示正方型 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <slot name="title"></slot>
      <div class="titian-grid titian-grid-divider">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="mine-to-pay"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待付款
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待收货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-comment"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待评价
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-refund"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            退货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="camera-point"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            拍照展示
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待收货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item titian-grid-item-square" style="width: 25%; padding-top: 25%;">
        <div class="titian-grid-item-content titian-grid-item-content-square undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
</ti-grid>
`;

exports[`渲染宫格组件 设置宫格间距离 1`] = `
<ti-grid>
  <mock:shadow-root>
    <div>
      <slot name="title"></slot>
      <div class="titian-grid titian-grid-divider titian-grid-gutter" style="padding-left: 4px;">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%; margin-top: 4px; padding-right: 4px;">
        <div class="titian-grid-item-content titian-grid-item-content-gutter undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="mine-to-pay"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待付款
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%; margin-top: 4px; padding-right: 4px;">
        <div class="titian-grid-item-content titian-grid-item-content-gutter undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-deliver"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待收货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%; margin-top: 4px; padding-right: 4px;">
        <div class="titian-grid-item-content titian-grid-item-content-gutter undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-receive"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待发货
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
  <ti-grid-item style="display: contents;">
    <mock:shadow-root>
      <div class="titian-grid-item" style="width: 25%; margin-top: 4px; padding-right: 4px;">
        <div class="titian-grid-item-content titian-grid-item-content-gutter undefined" style="flex-direction: column;">
          <div class="titian-grid-item-icon undefined">
            <ti-icon name="to-comment"></ti-icon>
          </div>
          <div class="titian-grid-item-text undefined">
            待评价
          </div>
        </div>
      </div>
    </mock:shadow-root>
  </ti-grid-item>
</ti-grid>
`;
